### PR TITLE
[NTDLL_APITEST] Add NtCreateFile FILE_APPEND_DATA test case

### PR DESCRIPTION
**Pull request by zecarlos1957**
[APITEST] Add NtCreateFile FILE_APPEND_DATA test case.

JIRA issue: [CORE-18789](https://jira.reactos.org/browse/CORE-18789)

### CODE REVIEW

**Comment by zecarlos1957** [comment]
Ntdll_apitest

File: modules/rostests/apitests/ntdll/NtWriteFile.c [hunk 1]
```diff
@@ -56,6 +59,87 @@ SizeOfSector(VOID)
...
+{
+    NTSTATUS Status;
+    HANDLE FileHandle;
+    IO_STATUS_BLOCK Iosb;
+    FILE_POSITION_INFORMATION FilePosInfo;
+    OBJECT_ATTRIBUTES ObjectAttributes;
+    CHAR chBuffer[256];
+    ULONG uDataLen;
+    UNICODE_STRING FileName;
+    RtlInitUnicodeString(&FileName, L"\\??\\C:\\Documents and Settings\\test.txt");
```
Comment (HBelusca) [inline modules/rostests/apitests/ntdll/NtWriteFile.c:72]: You should try investigate (some other apitests may do that) where to create that test file either in the root directory or somewhere in a temporary directory.

File: modules/rostests/apitests/ntdll/NtWriteFile.c [hunk 2]
```diff
@@ -56,6 +59,87 @@ SizeOfSector(VOID)
...
+                          FILE_OPEN_IF,
+                          FILE_NON_DIRECTORY_FILE | FILE_SYNCHRONOUS_IO_NONALERT |
+                            FILE_SEQUENTIAL_ONLY | FILE_WRITE_THROUGH,
+                          NULL,
+                          0);
+
+    lstrcpy((PCHAR)chBuffer, "Test NtCreateFile");
+    uDataLen = lstrlen(chBuffer)+1;
+
+    ZeroMemory(&Iosb, sizeof(Iosb));
```
Comment (HBelusca) [inline modules/rostests/apitests/ntdll/NtWriteFile.c:97]: Not needed

File: modules/rostests/apitests/ntdll/NtWriteFile.c [hunk 3]
```diff
@@ -56,6 +59,87 @@ SizeOfSector(VOID)
...
+
+    Status = NtWriteFile(FileHandle, NULL, NULL, NULL, &Iosb,
+                         chBuffer, uDataLen, NULL, NULL);
+     
+
+    ok_hex(Status, STATUS_SUCCESS);
+    Status = NtClose(FileHandle);
+    ok_hex(Status, STATUS_SUCCESS);
+    ZeroMemory(&Iosb, sizeof(Iosb));
```
Comment (HBelusca) [inline modules/rostests/apitests/ntdll/NtWriteFile.c:106]: ditto

File: modules/rostests/apitests/ntdll/NtWriteFile.c [hunk 4]
```diff
@@ -56,6 +59,87 @@ SizeOfSector(VOID)
...
+                                    sizeof(FilePosInfo),
+                                    FilePositionInformation);
+
+    ok_hex(Status, STATUS_SUCCESS);
+    ok(FilePosInfo.CurrentByteOffset.u.LowPart == -1, "Wrong File Pos %x. Expected -1\n", (UINT)FilePosInfo.CurrentByteOffset.u.LowPart);
+
+    lstrcpy((PCHAR)chBuffer, " FILE_APPEND_DATA");
+    uDataLen = lstrlen(chBuffer)+1;
+    RtlFreeUnicodeString(&FileName);
+    ZeroMemory(&Iosb, sizeof(Iosb));
```
Comment (HBelusca) [inline modules/rostests/apitests/ntdll/NtWriteFile.c:133]: ditto

File: modules/rostests/apitests/ntdll/NtWriteFile.c [hunk 5]
```diff
@@ -6,6 +6,9 @@
  */
 
 #include "precomp.h"
+#include <stdio.h>
+#include <fcntl.h>
+#include <io.h>
```
Comment (HBelusca) [inline modules/rostests/apitests/ntdll/NtWriteFile.c:11]: Not sure those 3 headers are necessary.

File: modules/rostests/apitests/ntdll/NtWriteFile.c [hunk 6]
```diff
@@ -316,4 +400,7 @@ START_TEST(NtWriteFile)
                                  &BufferSize,
                                  MEM_RELEASE);
     ok_hex(Status, STATUS_SUCCESS);
+
+    test_FileAppendData();
```
Comment (ThFabba) [inline modules/rostests/apitests/ntdll/NtWriteFile.c:404]: I'd recommend turning the current `START_TEST(NtWriteFile)` into a new function, then adding a new `START_TEST(NtWriteFile)` block that calls both functions.

**Comment by zecarlos1957** [comment]
Ntdll_apitest

**Comment by zecarlos1957** [comment]
TestOnWin11

**Comment by ThFabba** [comment]
Why is your test showing as succeeding in ReactOS and failing in Windows? That's the opposite of what it should be doing :)

**Comment by zecarlos1957** [comment]
On first failed test, there is a comment in the file:

    /* non-cached, too large -- fails to allocate MDL
     * Note: this returns STATUS_SUCCESS on Win7 -- higher MDL size limit */

Thats wy test got STATUS_SUCCESS
The second failed test is a strange behavior that i am investigating. The rostest is compiled to 32 bits and it is ruining on Wndows 11 64bits.
If i clean the Buffer with ZeroMemory(Buffer, sizeof(Buffer)), this solve the issue.

**Comment by zecarlos1957** [comment]
AppendData

**Pull request closed without merging**